_overviews/overview/a-taste-of-scala.md

```diff
@@ -0,0 +1,1329 @@
+- An `unapply` method is generated, which lets you use case classes in more ways in `match` expressions.
+- A `copy` method is generated in the class. This provides a way to clone an object while making updates to its values during the cloning process.
+- `equals` and `hashCode` methods are generated.
+- A default `toString` method is generated, which is helpful for debugging.
+
+This code demonstrates several case class features:
+
+```scala
+// define a case class
+case class Person(
+  name: String, 
+  vocation: String
+)
+
+// create an instance of the case class
+val p = Person("Reginald Kenneth Dwight", "Singer")
+
+// a good default toString method
+p                // Person = Person(Reginald Kenneth Dwight,Singer)
+
+// can access its fields, but they are immutable
+p.name           // "Reginald Kenneth Dwight"
+p.name = "Joe"   // error: can’t reassign a val field
+
+// when you need to make a change, use the `copy` method
+// to “update as you copy”
+val p2 = p.copy(name = "Elton John")
+p2               // Person = Person(Elton John,Singer)
+```
+
+See the Data Modeling sections of this Overview and the Reference documentation for many more details on case classes.
+
+
+
+## Scala methods
+
+<!--
+TODO: review Jonathan’s notes/categories about methods
+-->
+
+Scala classes, case classes, traits, enums, and objects can all contain methods. The general method syntax looks like this:
+
+```scala
+def methodName(param1: Type1, param2: Type2): ReturnType = 
+  // the method body
+  // goes here
+```
+
+Here are a few examples of that syntax:
+
+```scala
+def sum(a: Int, b: Int): Int = a + b
+def concatenate(s1: String, s2: String): String = s1 + s2
+```
+
+You don’t have to declare a method’s return type, so you can write those methods like this, if you prefer:
+
+```scala
+def sum(a: Int, b: Int) = a + b
+def concatenate(s1: String, s2: String) = s1 + s2
+```
+
+This is how you call those methods:
+
+```scala
+val x = sum(1, 2)
+val y = concatenate("foo", "bar")
+```
+
+Here’s an example of a multiline method:
+
+```scala
+def getStackTraceAsString(t: Throwable): String =
+  val sw = new StringWriter
+  t.printStackTrace(new PrintWriter(sw))
+  sw.toString
+```
+
+There are more things you can do with methods, such as providing default values for method parameters and using named parameters when calling methods. Those featurs are shown in the Data Modeling section of this Overview and in the Reference documentation.
+
+
+
+## First-class functions
+
+Scala has first-class support for functional programming, and included in that is first-class support for everything related to functions:
+
+- Top-level functions
+- Functions in objects
+- Lambdas
+- Higher-order functions (HOFs)
+
+<!-- TODO: when to say “function” vs when to say “method” -->
+<!-- TODO: show the `val` function syntax? -->
+
+
+###  Top-level functions
```

Comment:
I’m not sure we should have this section in the Taste page. Instead, I suggest a “Top-level definitions” page in the overview.

Reply:
Yes, if I did anything here I probably should have made a small “Top-level definitions” subsection. I had a mistake in my notes and I *thought* this was something that @b-studios suggested for this section, but I see that it was something I added in myself.

I did think that this was a nice new feature of Scala 3 to highlight. Do you think a “Top-level definitions” subsection is worth having in the Taste, or just have this as a later section in the Overview? If I don’t have that as a subsection here, I can refer readers to that section earlier on when I show the `@main def...` syntax.